src/rpc/mining.cpp

```diff
@@ -132,7 +131,7 @@ UniValue generateBlocks(std::shared_ptr<CReserveScript> coinbaseScript, int nGen
             LOCK(cs_main);
             IncrementExtraNonce(pblock, chainActive.Tip(), nExtraNonce);
         }
-        while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount && !CheckProofOfWork(pblock->GetMinedHash(nHeight + 1), pblock->nBits, Params().GetConsensus())) {
+        while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount && !CheckProofOfWork(pblock->GetMinedHash(Params().GetConsensus(), nHeight + 1), pblock->nBits, Params().GetConsensus())) {
```

Comment:
Let's move condition based on chain height to the CheckProofOfWork function
CheckProofOfWork(block, height)
{
 if (Scrypt2Period(height))
 {
 ... 
 }
...
}

Reply:
\+